packages/aws-cdk/lib/commands/init/init.ts

```
@@ -88,26 +256,54 @@ function pythonExecutable() {
 }
 const INFO_DOT_JSON = 'info.json';
 
+interface TemplateInitInfo {
+  readonly description: string;
+  readonly aliases?: string[];
+}
+
+enum TemplateType {
+  BUILTIN = 'builtin',
+  CUSTOM = 'custom',
+}
+
 export class InitTemplate {
   public static async fromName(templatesDir: string, name: string) {
     const basePath = path.join(templatesDir, name);
     const languages = await listDirectory(basePath);
     const initInfo = await fs.readJson(path.join(basePath, INFO_DOT_JSON));
-    return new InitTemplate(basePath, name, languages, initInfo);
+    return new InitTemplate(basePath, name, languages, initInfo, TemplateType.BUILTIN);
+  }
+
+  public static async fromPath(templatePath: string) {
+    const basePath = path.resolve(templatePath);
+
+    if (!await fs.pathExists(basePath)) {
+      throw new ToolkitError(`Template path does not exist: ${basePath}`);
+    }
+
+    const languages = await getLanguageDirectories(basePath);
+    const name = path.basename(basePath);
+
+    return new InitTemplate(basePath, name, languages, null, TemplateType.CUSTOM);
   }
 
   public readonly description: string;
   public readonly aliases = new Set<string>();
+  public readonly templateType: TemplateType;
 
   constructor(
     private readonly basePath: string,
     public readonly name: string,
     public readonly languages: string[],
-    initInfo: any,
+    initInfo: TemplateInitInfo | null,
+    templateType: TemplateType,
   ) {
-    this.description = initInfo.description;
-    for (const alias of initInfo.aliases || []) {
-      this.aliases.add(alias);
+    this.description = initInfo?.description ?? 'Custom template';
```

where is description being used?

---

It's not used. It's only used for built-in templates that specify a description in their info.json file. In the existing code (main branch), the description is then used to show what each built-in template does when a user runs `cdk init --list`. This doesn't apply to custom templates since they don't have info.json files that define a description and also usage instructions for them are not displayed through `cdk init --list` so I can remove it. I just had it in there to keep custom templates properties consistent with built-in template properties so I had custom templates default to a hardcoded string i made.

Fixing the code so that custom template's don't use descriptions at all or have any kind of hardcoded description that I define either.